resources/public/api/conf/1.0/checkpoints_list.yaml

```diff
@@ -0,0 +1,49 @@
+get:
+  $ref: "./checkpoints_list.yaml"
+  summary: List checkpoints
+  description: |
+    This endpoint allows a developer to get all the checkpoints created for the vendor id. 
+    A National Insurance number can be provided to to filter the results based on National Insurance number.
+  security:
+    - User-Restricted:
+        - write:self-assessment
+  parameters:
+    - $ref: './common/headers.yaml#/components/parameters/acceptHeader'
+    - $ref: './common/headers.yaml#/components/parameters/authorizationWriteScopeHeader'
+    - $ref: './common/queryParameters.yaml#/components/parameters/ninoCreateCheckpoint'
+
+  responses:
+    200:
+      description: Success
+      headers:
+        X-CorrelationId:
+          $ref: './common/headers.yaml#/components/parameters/correlationId'
+      content:
+        application/json:
+          schema:
+            $ref: './schemas/checkpoints/list_response.json'
+          examples:
+            example-response:
+              description: Example Response
+              value:
+                $ref: './examples/checkpoints/list_response.json'
+
+    400:
+      description: Bad Request
+      content:
+        application/json:
+          schema:
+            $ref: './schemas/error_response.yaml'
+          examples:
+            FORMAT_NINO:
+              $ref: './common/errors.yaml#/components/examples/formatNino'
+
+    403:
+      description: Forbidden
+      content:
+        application/json:
+          schema:
+            $ref: './schemas/error_response.yaml'
+          examples:
+            CLIENT_OR_AGENT_NOT_AUTHORISED:
+              $ref: './common/errors.yaml#/components/examples/clientOrAgentNotAuthorised'
```

Comment:
I think we also need to add a "404" error. The business spec currently misses this error out, but it is included in the tech spec.